pkg/operatorloglevel/loglevel_controller.go

```diff
@@ -0,0 +1,179 @@
+package operatorloglevel
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog"
+)
+
+type operatorLogLevelNormalizer struct {
+	dynamicClient dynamic.Interface
+}
+
+// NewLogLevelNormalizer normalizes the log level in operator custom resource
+// if it's value differs from supported values
+func NewLogLevelNormalizer(dynamicClient dynamic.Interface,
+	operatorClient operatorv1helpers.OperatorClient,
+	recorder events.Recorder) factory.Controller {
+	c := operatorLogLevelNormalizer{
+		dynamicClient: dynamicClient,
+	}
+
+	return factory.New().
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ResyncEvery(5*time.Minute).
+		ToController("OperatorLogLevelNormalizer", recorder)
+}
+
+// sync runs periodically and makes sure that the log level field values in operator
+// spec does not differ from supported values.
+func (c *operatorLogLevelNormalizer) sync(ctx context.Context, syncCtx factory.SyncContext) error {
```

Comment:
please create a follow-up BZ that we can remove this controller again one release after we enabled stricter validation.